From b00a562ffcd3c5a3d50227ccb9b11b9c2c13087b Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Mon, 21 Oct 2019 11:44:20 +0300
Subject: [PATCH 19/39] <sys/tty.h>: Remove explicit dependency on
 <sys/malloc.h>.

 Only two kernel sources makes use of M_TTYS.
---
 sys/dev/serial/sio/sio_pccard.c | 1 +
 sys/kern/tty.c                  | 2 +-
 sys/sys/tty.h                   | 6 ++----
 3 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/sys/dev/serial/sio/sio_pccard.c b/sys/dev/serial/sio/sio_pccard.c
index 0fad8486e9..1d17b53a96 100644
--- a/sys/dev/serial/sio/sio_pccard.c
+++ b/sys/dev/serial/sio/sio_pccard.c
@@ -31,6 +31,7 @@
 #include <sys/systm.h>
 #include <sys/tty.h>
 #include <sys/proc.h>
+#include <sys/malloc.h>
 #include <sys/module.h>
 #include <sys/kernel.h>
 #include <sys/bus.h>
diff --git a/sys/kern/tty.c b/sys/kern/tty.c
index 5443f977ae..cf52425af9 100644
--- a/sys/kern/tty.c
+++ b/sys/kern/tty.c
@@ -77,6 +77,7 @@
 #include <sys/systm.h>
 #include <sys/uio.h>
 #include <sys/filio.h>
+#include <sys/malloc.h>
 #include <sys/proc.h>
 #include <sys/priv.h>
 #include <sys/tty.h>
@@ -91,7 +92,6 @@
 #include <sys/signalvar.h>
 #include <sys/signal2.h>
 #include <sys/resourcevar.h>
-#include <sys/malloc.h>
 #include <sys/filedesc.h>
 #include <sys/sysctl.h>
 #include <sys/thread2.h>
diff --git a/sys/sys/tty.h b/sys/sys/tty.h
index de90b5e5a7..a0f42a05bd 100644
--- a/sys/sys/tty.h
+++ b/sys/sys/tty.h
@@ -231,11 +231,9 @@ struct speedtab {
 
 #ifdef _KERNEL
 
-#ifndef _SYS_MALLOC_H_
-#include <sys/malloc.h>
-#endif
-
+#ifdef MALLOC_DECLARE
 MALLOC_DECLARE(M_TTYS);
+#endif
 
 extern	struct tty *constty;	/* Temporary virtual console. */
 
-- 
2.23.0

